plugins/TauMVAProducer.cc

```diff
@@ -0,0 +1,371 @@
+	totiso0p4.push_back(totiso0p4_);
+	trackiso.push_back(trackiso_);
+	nearphopt.push_back(photonIndex_ > -1 ? pfcands_->at(photonIndex_).pt() : -1.0);
+	nearphoeta.push_back(photonIndex_ > -1 ? pfcands_->at(photonIndex_).eta() : -1.0);
+	nearphophi.push_back(photonIndex_ > -1 ? pfcands_->at(photonIndex_).phi() : -1.0);
+	nearestTrkDR.push_back(nearesttrkdr_);
+	contJetIndex.push_back(jetIndex_);
+
+	//Store in the tree jet pt -> 20 so that in the next round we can go that low
+	jetmatch = jetIndex_ > -1 && jet->pt() >= 20.0 && fabs(jet->eta()) < 2.4;
+	contjetdr_ = jetmatch ? deltaR(pfc->p4(), jet->p4()) : -1.0;
+	contjetcsv_ = jetmatch ? jet->bDiscriminator("pfCombinedInclusiveSecondaryVertexV2BJetTags") : -1.0;
+	
+	
+	contJetDR.push_back(contjetdr_);
+	contJetCSV.push_back(contjetcsv_);
+	
+	edm::Ptr<reco::Candidate> c = pfcands_->ptrAt(ic);
+	selCandPf->push_back(c);
+    }
+
+    auto out = std::make_unique<nanoaod::FlatTable>(selCandPf->size(), "pfcand", false); 
+    //out->setDoc("save pfcand and tau mva variables");
+    
+    //out->addColumn<float>("pt", 			pt	  , "pfcand info pT" 		  , nanoaod::FlatTable::FloatColumn,10);
+    //out->addColumn<float>("eta", 			eta	  , "pfcand info Eta"		  , nanoaod::FlatTable::FloatColumn,10);
+    //out->addColumn<float>("phi", 			phi	  , "pfcand info Phi"		  , nanoaod::FlatTable::FloatColumn,10);
+    //out->addColumn<float>("mass", 			mass	  , "pfcand info Mass"		  , nanoaod::FlatTable::FloatColumn,10);
+    out->addColumn<float>("dz", 		dz	  , "pfcand info dz"			  , nanoaod::FlatTable::FloatColumn,10);
+    out->addColumn<float>("fromPV", 		fromPV	  ,"pfcand info from Primary Vertex"  	  , nanoaod::FlatTable::FloatColumn,10);
+    out->addColumn<float>("chiso0p1", 		chiso0p1, "charged hadron isolation with R = 0.1" , nanoaod::FlatTable::FloatColumn,10);
+    out->addColumn<float>("chiso0p2", 		chiso0p2, "charged hadron isolation with R = 0.2" , nanoaod::FlatTable::FloatColumn,10);
+    out->addColumn<float>("chiso0p3", 		chiso0p3, "charged hadron isolation with R = 0.3" , nanoaod::FlatTable::FloatColumn,10);
+    out->addColumn<float>("chiso0p4", 		chiso0p4, "charged hadron isolation with R = 0.4" , nanoaod::FlatTable::FloatColumn,10);
+    out->addColumn<float>("totiso0p1", 		totiso0p1, "total hadron isolation with R = 0.1"  , nanoaod::FlatTable::FloatColumn,10);
+    out->addColumn<float>("totiso0p2", 		totiso0p2, "total hadron isolation with R = 0.2"  , nanoaod::FlatTable::FloatColumn,10);
+    out->addColumn<float>("totiso0p3", 		totiso0p3, "total hadron isolation with R = 0.3"  , nanoaod::FlatTable::FloatColumn,10);
+    out->addColumn<float>("totiso0p4", 		totiso0p4, "total hadron isolation with R = 0.4"  , nanoaod::FlatTable::FloatColumn,10);
+    out->addColumn<float>("trackiso", 		trackiso,  "individual track isolation"		  , nanoaod::FlatTable::FloatColumn,10);
+    out->addColumn<float>("nearphopt", 		nearphopt, "pT of nearest photon to track" 	  , nanoaod::FlatTable::FloatColumn,10);
```

Comment:
What is the definition of photon here? How do you use it? Can you just use the photon stored by NanoAOD directly and link the index? Since Photon are already stored in NanoAOD, it won't duplicate the information.

Reply:
I might need to take it back, as the Photon only stored with pt > 5GeV in NanoAOD. So these seems necessary.